clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryClient.java

```diff
@@ -45,17 +58,84 @@ public DataRecoveryMonitor getMonitor() {
   }
 
+  public Map<String, Pair<Boolean, String>> getRepushViability(
+      Set<String> storesList,
+      StoreRepushCommand.Params params) {
+    Map<String, Pair<Boolean, String>> ret = new HashMap<>();
+    String url = params.getUrl();
+    ControllerClient cli = params.getPCtrlCliWithoutCluster();
+    LocalDateTime timestamp = params.getTimestamp();
+    String destFabric = params.getDestFabric();
+    for (String s: storesList) {
+      try {
+        String clusterName = cli.discoverCluster(s).getCluster();
+        if (clusterName == null) {
+          ret.put(s, Pair.of(false, "unable to discover cluster for store (likely invalid store name)"));
+          continue;
+        }
+        ControllerClient parentCtrlCli = buildControllerClient(clusterName, url, params.getSSLFactory());
+        StoreHealthAuditResponse storeHealthInfo = parentCtrlCli.listStorePushInfo(s, false);
+        Map<String, RegionPushDetails> regionPushDetails = storeHealthInfo.getRegionPushDetails();
+        if (!regionPushDetails.containsKey(destFabric)) {
+          ret.put(s, Pair.of(false, "nothing to repush, store version 0"));
+          continue;
+        }
+        String latestTimestamp = regionPushDetails.get(destFabric).getPushStartTimestamp();
+        LocalDateTime latestPushStartTime = LocalDateTime.parse(latestTimestamp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+
+        if (latestPushStartTime.isBefore(timestamp)) {
+          ret.put(s, Pair.of(false, "input timestamp earlier than latest push"));
+          continue;
+        }
+
+        MultiStoreStatusResponse storeStatusResponse = parentCtrlCli.getFutureVersions(clusterName, s);
+        if (storeStatusResponse.getStoreStatusMap().containsKey(destFabric)) {
+          ret.put(s, Pair.of(false, "ongoing repush"));
+        } else {
+          ret.put(s, Pair.of(true, ""));
+        }
+      } catch (VeniceException e) {
+        ret.put(s, Pair.of(false, "VeniceHttpException " + e.getErrorType().toString()));
+      }
+    }
+    return ret;
+  }
+
+  public ControllerClient buildControllerClient(String clusterName, String url, Optional<SSLFactory> sslFactory) {
+    return new ControllerClient(clusterName, url, sslFactory);
+  }
+
   public void execute(DataRecoveryParams drParams, StoreRepushCommand.Params cmdParams) {
     Set<String> storeNames = drParams.getRecoveryStores();
-    if (storeNames == null || storeNames.isEmpty()) {
-      LOGGER.warn("store list is empty, exit.");
-      return;
+    Map<String, Pair<Boolean, String>> pushMap = getRepushViability(storeNames, cmdParams);
+    Set<String> filteredStoreNames = new HashSet<>();
+
+    for (Map.Entry<String, Pair<Boolean, String>> e: pushMap.entrySet()) {
+      if (e.getValue().getLeft() == true) {
+        filteredStoreNames.add(e.getKey());
+      }
     }
-    if (!drParams.isNonInteractive && !confirmStores(storeNames)) {
-      return;
+
+    if (filteredStoreNames != null && !filteredStoreNames.isEmpty()) {
```

Comment:
Nit, we can skip `filteredStoreNames != null` check, because it cannot be null, (allocated several lines before @ L111)